### DISCUSSION TITLE
Is it possible to send alerts when processing errors occure

### DISCUSSION THREAD

**Discussion by archfz** [jertel/elastalert2#715] [Q&A] [answered]
Like for example when this issue occurs https://github.com/jertel/elastalert2/discussions/383 it would be really nice if an alert would be sent to notify that the rule cannot run and thus we know there is an issue. Otherwise one might thing everything is ok.

**Comment by nsano-rururu** [comment, marked as answer]
https://github.com/Yelp/elastalert/issues/1892#issuecomment-419583359
>This would be a good feature to add. There's a `notify_email` option, but this generally only sends you something if an uncaught exception occurs, if a rule fails to load, or if Elasticsearch is uncontactable when ElastAlert first starts.
>
>For some types of errors, like timeouts or errors in the filter, you can actually setup a "meta" elastalert for the error logs (elastalert_status _type: error, or elastalert_status_error in ES 6). But, of course, this won't work in the case when Elasticsearch is completely down.

**Reply by archfz** [reply]
So I suppose it's a feature request? I see it was never implemented and if I understand correctly that notify_email doesn't work for this purpose. Also I would prefer not email but discord.

**Reply by nsano-rururu** [reply]
See past issues on the original yelp / elastalert. I don't have the features you want and I have no plans to implement them. Pull requests are welcome.
https://github.com/Yelp/elastalert/issues/1892

**Reply by archfz** [reply]
I have found that `elastalert_status_error` index actually had an instance of the error. I wonder if one could setup an alert on this index. The issue is that I am still unsure why the `circuit_breaking_exception` was occurring for me, as I was not receiving multiple types of alerts or actually maybe even no types, so maybe in such case even a rule over this index would fail. My issue was solved only after deleting a lot of indices that were not properly configured to roll over. But even alert rules targeting indicies that were properly rolling over were not working. 
![image](https://user-images.githubusercontent.com/13706730/154261465-f38bca96-680c-487f-9e3c-1292006559f2.png)

**Reply by archfz** [reply]
Also interestingly this error in the index was logged only once while in the service logs it was constantly reproducing for many days straight.

**Reply by archfz** [reply]
And never mind, it got only logged after fixes were applied, not as soon as the issue reproduced.

**Reply by nsano-rururu** [reply]
There seems to be some people who devise and monitor whether elastalert is working.
https://github.com/mitodl/salt-ops/pull/1181
>This adds the Datadog process.d collector to minions running elastalert, so we can check that elastalert is running.

**Reply by archfz** [reply]
Is that a service in addtion? Like a service checking another service that is checking the final services? That is not great.

**Reply by jertel** [reply]
`Monitoring the monitor` is an essential part of systems operations. For best results, the secondary monitor should be isolated from the primary monitoring system (ElastAlert 2) so that in the event of a catastrophic failure, the secondary monitor will still be online and able to detect anomalies. For example, suppose ElastAlert 2 is running as a pod inside of an Azure Kubernetes cluster (AKS) that also runs the Elasticsearch pods. What happens if Azure has an outage the completely takes down that AKS cluster? If you had a secondary monitor running inside that same cluster monitoring processes, or even ES index data for lapses, then that secondary monitor is also offline and you would never be notified of the outage. However, if you run that secondary monitor outside of the AKS platform, such as in Amazon or Google, then it can detect the outage. One way to detect the outage is to have ElastAlert 2 trigger an alert every minute or so, and that alert reaches out to the secondary system via an HTTP post, or email, etc. The secondary system can detect that ElastAlert 2 hasn't reach out in X minutes, and therefore it will send it's own Slack or Email message to make operators aware of the outage.

Edit: To clarify, I'm not disagreeing with you, I'm just pointing out that a scenario often exists where it would be important to monitor that ElastAlert 2 is doing its job, particularly when ElastAlert 2 is running in the same environment as ES. Your original suggestion of enhancing ElastAlert 2 to support alerting when fault is detected is a good idea and a PR would be welcome for that. This enhancement would be beneficial If ElastAlert 2 is already running in an isolated environment outside of the ES environment.
